ark/jsonschema/__tests__/string.test.ts

```diff
@@ -0,0 +1,79 @@
+import { attest, contextualize } from "@ark/attest"
+import { parseJsonSchema } from "@ark/jsonschema"
+
+contextualize(() => {
+	it("type string", () => {
+		const t = parseJsonSchema({ type: "string" })
+		attest(t.json).snap({ domain: "string" })
+	})
+
+	it("maxLength (positive)", () => {
+		const tMaxLength = parseJsonSchema({
+			type: "string",
+			maxLength: 5
+		})
+		attest(tMaxLength.json).snap({
+			domain: "string",
+			maxLength: 5
+		})
+	})
+
+	it("maxLength (negative)", () => {
+		const maxLength = -5
+		attest(() =>
+			parseJsonSchema({
+				type: "string",
+				maxLength
+			})
+		).throws(
+			`TraversalError: maxLength must be non-negative (was ${maxLength})`
```

Comment:
Can remove the custom error here as well.

Reply:
Once again, this isn't actually a custom error. It's the default error that ArkType provides.

We just explicitly check for the error since the JSON Schema spec states this scenario must explicitly error.

Reply:
Marking as resolved. Can be further discussed on https://github.com/arktypeio/arktype/pull/1159#discussion_r2015124232